app/models/pending_purchase.rb

```diff
@@ -3,4 +3,17 @@ class PendingPurchase < ActiveRecord::Base
 
   belongs_to :user
   belongs_to :book
+
+  def total_price
+    price_at_purchase * quantity
+  end
+
+  def was_purchased
+    self.purchased = true
+    save
+  end
+
+  def self.pending
+    PendingPurchase.all.reject { |pending_purchase| pending_purchase.purchased == true }
```

Comment:
Good job switching this over. Two notes about the new code:

1. Change `PendingPurchase.all.reject ...` to `PendingPurchase.where(purchased: false)`. `reject` will iterate over all records.
2. Change this to a [scope](http://api.rubyonrails.org/classes/ActiveRecord/Scoping/Named/ClassMethods.html)